torch/_functorch/aot_autograd.py

```diff
@@ -468,9 +468,12 @@ def create_aot_dispatcher_function(
         enable_python_dispatcher() if shape_env is not None else nullcontext()
     )
 
-    with torch.autograd.set_multithreading_enabled(
-        False
-    ), preserve_rng_state(), fake_mode, python_dispatcher_mode, PhiloxStateTracker():
+    with torch.autograd.set_multithreading_enabled(False), \
+            preserve_rng_state(), \
+            fake_mode, \
+            python_dispatcher_mode, \
+            PhiloxStateTracker(), \
+            torch._C._DisableTorchFunctionNonInfraSubclass():
```

Comment:
It seems to me that this should be toggleable? Because if I directly call aot autograd (e.g., from non-strict export), I would *not* expect the torch function subclass to have already be inlined and therefore I really should not disable it.

Overall, I find doing the disabling here inside AOTAutograd a bit smelly, but I'm not sure what to do about it. Potentially, it could be that *Dynamo* should be responsible for disabling non-infra torch function subclasses before calling into the backend, as a way of indicating that it has already inlined through them. So you'd move this over to output_graph.py or something.